src/Symfony/Component/Process/Tests/ProcessTest.php

```diff
@@ -133,6 +133,34 @@ public function testStopWithTimeoutIsActuallyWorking()
         $this->assertLessThan(15, microtime(true) - $start);
     }
 
+    public function testWaitUntilSpecificOutput()
+    {
+        $p = $this->getProcess(array(self::$phpBin, __DIR__.'/KillableProcessWithOutput.php'));
+        $p->start();
+
+        $start = microtime(true);
+
+        $completeOutput = '';
+        $p->waitUntil(function ($type, $output) use (&$completeOutput) {
+            $completeOutput .= $output;
+            if (false !== strpos($output, 'One more')) {
+                return true;
+            }
+
+            return false;
+        });
+        $p->stop();
+
+        $this->assertEquals("First iteration output\nSecond iteration output\nOne more iteration output\n", $completeOutput);
+
+        if ('\\' === \DIRECTORY_SEPARATOR) {
+            // Windows is slower
+            $this->assertLessThan(15, microtime(true) - $start);
```

Comment:
you should assert that before the assertion on the output. Or at least measure the time before that point. Otherwise, your measurement includes the time it takes for PHPUnit to compare the strings.

Reply:
Nice catch. Fixed.